controller/query.go

```diff
@@ -0,0 +1,147 @@
+			Title:   strings.TrimSpace(ctx.Payload.Data.Attributes.Title),
+			Creator: *currentUserIdentityID,
+		}
+		// Parse fields to make sure that query is valid
+		_, _, err := search.ParseFilterString(ctx, q.Fields)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"space_id": ctx.SpaceID,
+				"fields":   q.Fields,
+			}, "unable to parse the query fields")
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		err = appl.Queries().Create(ctx, &q)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		res := &app.QuerySingle{
+			Data: ConvertQuery(appl, ctx.Request, q),
+		}
+		ctx.ResponseData.Header().Set("Location", rest.AbsoluteURL(ctx.Request, app.LabelHref(ctx.SpaceID, res.Data.ID)))
+		return ctx.Created(res)
+	})
+}
+
+// ConvertQuery converts from internal to external REST representation
+func ConvertQuery(appl application.Application, request *http.Request, q query.Query) *app.Query {
+	queryType := query.APIStringTypeQuery
+	spaceID := q.SpaceID.String()
+	relatedURL := rest.AbsoluteURL(request, app.LabelHref(spaceID, q.ID))
+	creatorID := q.Creator.String()
+	userType := APIStringTypeUser
+	relatedCreatorLink := rest.AbsoluteURL(request, fmt.Sprintf("%s/%s", usersEndpoint, creatorID))
+	// spaceRelatedURL := rest.AbsoluteURL(request, app.SpaceHref(spaceID))
+	appQuery := &app.Query{
+		Type: queryType,
+		ID:   &q.ID,
+		Attributes: &app.QueryAttributes{
+			Title:     q.Title,
+			Fields:    q.Fields,
+			CreatedAt: &q.CreatedAt,
+			UpdatedAt: &q.UpdatedAt,
+		},
+		Links: &app.GenericLinks{
+			Self:    &relatedURL,
+			Related: &relatedURL,
+		},
+		Relationships: &app.QueryRelations{
+			Creator: &app.RelationGeneric{
+				Data: &app.GenericData{
+					Type: &userType,
+					ID:   &creatorID,
+					Links: &app.GenericLinks{
+						Related: &relatedCreatorLink,
+					},
+				},
+			},
+		},
+	}
+	return appQuery
+}
+
+// ConvertQueries from internal to external REST representation
+func ConvertQueries(appl application.Application, request *http.Request, queries []query.Query) []*app.Query {
+	var ls = []*app.Query{}
+	for _, q := range queries {
+		ls = append(ls, ConvertQuery(appl, request, q))
+	}
+	return ls
+}
+
+// List runs the list action.
+func (c *QueryController) List(ctx *app.ListQueryContext) error {
+	currentUserIdentityID, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	return application.Transactional(c.db, func(appl application.Application) error {
+		queries, err := appl.Queries().ListByCreator(ctx, ctx.SpaceID, *currentUserIdentityID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		res := &app.QueryList{}
+		res.Data = ConvertQueries(appl, ctx.Request, queries)
+		return ctx.OK(res)
+	})
+}
+
+// Show runs the show action.
+func (c *QueryController) Show(ctx *app.ShowQueryContext) error {
+	// QueryController_Show: start_implement
+
+	// Put your logic here
```

Comment:
This is required for `QueryHref` to work.

Reply:
yes that was todo part, added in https://github.com/fabric8-services/fabric8-wit/pull/1824/commits/2c7f2d8b82dfd68ccfa159a7e6373bfe991266c4